spark3/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -133,12 +134,25 @@ public void pruneColumns(StructType requestedSchema) {
     this.requestedProjection = requestedSchema;
   }
 
+  public SparkScanBuilder ignoreResiduals() {
```

Comment:
I decided to make it configurable so that we can reuse it for merge-on-read too.